src/ucp/api/ucp.h

```diff
@@ -4497,6 +4527,17 @@ ucs_status_t ucp_dt_create_generic(const ucp_generic_dt_ops_t *ops, void *contex
  */
 void ucp_dt_destroy(ucp_datatype_t datatype);
 
+/**
+ * @ingroup UCP_DATATYPE
+ * @brief Query attributes of a datatype.
+ *
+ * When UCP_DATATYPE_ATTR_FIELD_PACKED_SIZE is set in field_mask of @a attr, the
+ * the field packed_size is set to the packed size (bytes) of the datatype.
+ * If the size could not be determined, 0 is returned.
+ * @note The size of a generic datatype is not available before it has been used
```

Comment:
also what value should we return for iov?

Reply:
That's right, the API does not work for IOV data and since there is no public API that creates an IOV datatype, we cannot know the size of the vector. Thus, it falls into the category "size could not be determined", but it should be noted in the description separately.